cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -256,16 +256,9 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
         result.setOwnerReference(kafkaAssembly);
         KafkaClusterSpec kafkaClusterSpec = kafkaAssembly.getSpec().getKafka();
         result.setReplicas(kafkaClusterSpec.getReplicas());
-        String version = kafkaClusterSpec.getVersion();
-        if (version == null) {
-            version = versions.defaultVersion().version();
-        }
-        String image = kafkaClusterSpec.getImage();
-        if (image == null) {
-            image = IMAGE_MAP.get(version);
-        }
+        String image = versions.kafkaImage(kafkaClusterSpec.getImage(), kafkaClusterSpec.getVersion());
         if (image == null) {
-            image = IMAGE_MAP.get(versions.defaultVersion().version());
+            throw new InvalidResourceException("Version is not supported");
```

Comment:
how does defaulting to an image works now?

Reply:
It's explained in #1126. Please comment there if it's still not clear.